llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1182,9 +1182,10 @@ void AsmPrinter::emitStackSizeSection(const MachineFunction &MF) {
   OutStreamer->PopSection();
 }
 
-static bool needFuncLabelsForEHOrDebugInfo(const MachineFunction &MF) {
-  MachineModuleInfo &MMI = MF.getMMI();
-  if (!MF.getLandingPads().empty() || MF.hasEHFunclets() || MMI.hasDebugInfo())
+static bool needFuncLabelsForEHOrDebugInfo(const MachineFunction &MF,
+                                           bool IncludeDebugInfo = true) {
+  if (!MF.getLandingPads().empty() || MF.hasEHFunclets() ||
+      (IncludeDebugInfo && MF.getMMI().hasDebugInfo()))
     return true;
```

Comment:
Maybe nicer to split it?

```suggestion
static bool needFuncLabelsForEHOrDebugInfo(const MachineFunction &MF) {
  return MF.getMMI().hasDebugInfo() || needFuncLabelsForEH(MF);
}

static bool needFuncLabelsForEH(const MachineFunction &MF) {
  if (!MF.getLandingPads().empty() || MF.hasEHFunclets())
    return true;
```

Or you can do:

```suggestion
static bool needFuncLabelsForEH(const MachineFunction &MF) {
  return needFuncLabelsForEHOrDebugInfo(MF, false);
}

static bool needFuncLabelsForEHOrDebugInfo(const MachineFunction &MF,
                                           bool IncludeDebugInfo = true) {
  if (!MF.getLandingPads().empty() || MF.hasEHFunclets() ||
      (IncludeDebugInfo && MF.getMMI().hasDebugInfo()))
    return true;
```

I guess, though I don't like that as much (but still better than having an `AOrB` where you pass `/*IncludeB=*/false` in uses that aren't wrappers IMO)

Reply:
There are only two uses, so splitting it sounds reasonable.